dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-to-fee-record-payment-groups.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  FeeRecordPaymentEntityGroup,
+  calculateTotalCurrencyAndAmount,
+  getFeeRecordPaymentEntityGroupReconciliationData,
+  getFeeRecordPaymentEntityGroupStatus,
+} from '../../../../../helpers';
+import { mapPaymentEntityToPayment } from '../../../../../mapping/payment-mapper';
+import { mapFeeRecordEntityToFeeRecord } from '../../../../../mapping/fee-record-mapper';
+import { FeeRecordPaymentGroup } from '../../../../../types/utilisation-reports';
+
+/**
+ * Maps the fee record payment entity groups to the fee record payment groups
+ * @param feeRecordPaymentEntityGroups - The fee record payment entity groups
+ * @returns The fee record payment groups
+ */
+export const mapToFeeRecordPaymentGroups = async (feeRecordPaymentEntityGroups: FeeRecordPaymentEntityGroup[]): Promise<FeeRecordPaymentGroup[]> =>
+  await Promise.all(
+    feeRecordPaymentEntityGroups.map(async (group) => {
+      const groupStatus = getFeeRecordPaymentEntityGroupStatus(group);
+      const groupReconciliationData = await getFeeRecordPaymentEntityGroupReconciliationData(group);
+
+      if (group.payments.length === 0) {
+        // If there are no payments, there is only one fee record in the group
+        if (group.feeRecords.length !== 1) {
+          throw new Error('Fee record payment entity group cannot have more than one fee record if there are no payments');
```

Comment:
Is this handled gracefully - wondering if this error is thrown, will the central API still be running?

Reply:
Yes - this should always be called inside a `try-catch` and the caller needs to worry about handling the error correctly. However, the docstring of this function was missing the error tag, so I've added that